pkg/controller/machinepool/awsactuator_test.go

```diff
@@ -567,10 +571,116 @@ func TestAWSActuator(t *testing.T) {
 			},
 			expectedErr: true,
 		},
+		{
+			name:              "machinepool has AdditionalSecurityGroupIDs",
+			clusterDeployment: testClusterDeployment(),
+			machinePool: func() *hivev1.MachinePool {
+				pool := testMachinePool()
+				pool.Spec.Platform.AWS.AdditionalSecurityGroupIDs = []string{
+					"sg-one", "sg-two",
+				}
+				pool.Spec.Platform.AWS.Zones = []string{"zone1", "zone2"}
+				pool.Spec.Platform.AWS.Subnets = []string{"subnet-zone1", "subnet-zone2", "pubSubnet-zone1", "pubSubnet-zone2"}
+				return pool
+			}(),
+			masterMachine: testMachine("master0", "master"),
+			mockAWSClient: func(client *mockaws.MockClient) {
+				gomock.InOrder(
+					mockDescribeSubnets(client, []string{"zone1", "zone2"},
+						[]string{"subnet-zone1", "subnet-zone2"}, []string{"pubSubnet-zone1", "pubSubnet-zone2"}, "vpc-1"),
+				)
+				mockDescribeRouteTables(client,
+					map[string]bool{
+						"subnet-zone1":    false,
+						"subnet-zone2":    false,
+						"pubSubnet-zone1": true,
+						"pubSubnet-zone2": true,
+					},
+					"vpc-1")
+			},
+			expectedMachineSetReplicas: map[string]int64{
+				generateAWSMachineSetName("zone1"): 2,
+				generateAWSMachineSetName("zone2"): 1,
+			},
+			expectedAMI: &machineapi.AWSResourceReference{
+				ID: pointer.String(testAMI),
+			},
+			expectedSGFilters: []machineapi.Filter{
+				{
+					Name:   "tag:Name",
+					Values: []string{"foo-12345-worker-sg"},
+				},
+			},
+			expectedSGIDs: []string{"sg-one", "sg-two"},
+		}, {
+			name:              "ExtraWorkerSecurityGroup and AdditionalSecurityGroupIDs are mutually exclusive",
+			clusterDeployment: testClusterDeployment(),
+			machinePool: func() *hivev1.MachinePool {
+				pool := testMachinePool()
+				pool.Annotations = map[string]string{
+					constants.ExtraWorkerSecurityGroupAnnotation: "extra-security-group",
+				}
+				pool.Spec.Platform.AWS.AdditionalSecurityGroupIDs = []string{"additional-security-group"}
+				return pool
+			}(),
+			masterMachine: testMachine("master0", "master"),
+			expectedCondition: &hivev1.MachinePoolCondition{
+				Type:   hivev1.UnsupportedConfigurationMachinePoolCondition,
+				Status: corev1.ConditionTrue,
+				Reason: "SecurityGroupOptionConflict",
+			},
+		},
 	}
 
-	for _, test := range tests {
+	validateAWSMachineSets := func(t *testing.T, mSets []*machineapi.MachineSet, test ttype) {
```

Comment:
NTR: This chunk is mostly refactor. I was going to have to add another param to the func, which already had a param list a mile long. All but the first two were `test.something`, so I wanted to just be able to pass in `test`. In order to do so, I had to de-anonymize the test struct and then scope this func so I could ref it. (I could have pulled the `ttype` def out of the test function, but that seemed more wronger.)

More below...